src/ralph/conf.py

```diff
@@ -4,10 +4,18 @@
 from pathlib import Path
 from typing import Literal, Union
 
-from click import get_app_dir
+try:
+    from click import get_app_dir
+except ImportError:
+    # If we use Ralph as a library and Click is not installed, we consider the
+    # application directory to be the current directory. For non-CLI usage, it
+    # as no consequences.
+    from unittest.mock import Mock
+
+    get_app_dir = Mock(return_value=".")
```

Comment:
I'm not confortable with using mocks for anything else than tests. But maybe it's close minded. 
Is it better to write 2 classes and return either one depending on the case? Or set APP_DIR in the `__new__` method?

Reply:
> I'm not confortable with using mocks for anything else than tests. But maybe it's close minded.

Reply:
> I'm not confortable with using mocks for anything else than tests. But maybe it's close minded.

I was also surprised to read this, but it seems to be a common pattern: https://realpython.com/python-import/#handle-missing-packages-use-a-mock-instead